src/main/java/org/example/springproject/controller/BookController.java

```diff
@@ -52,4 +53,9 @@ public BookDto updateBook(@PathVariable @Positive Long id,
     public void deleteBook(@PathVariable @Positive Long id) {
         bookService.deleteById(id);
     }
+
+    @GetMapping("/search")
+    public List<BookDto> search(BookSearchParametersDto bookSearchParametersDto) {
```

Comment:
```suggestion
    public List<BookDto> search(BookSearchParametersDto bookSearchParametersDto, @ParameterObject @PageableDefault Pageable pageable) {
```

Reply:
Pagination topic is not yet completed. I will add this functional later